ledger-service/http-json/src/main/scala/com/digitalasset/http/ContractsService.scala

```diff
@@ -184,69 +214,115 @@ class ContractsService(
         status = StatusCodes.BadRequest
       )
     } else {
-      val source = daoAndFetch.cata(
-        x => searchDb(x._1, x._2)(jwt, parties, resolvedTemplateIds, queryParams),
-        searchInMemory(jwt, parties, resolvedTemplateIds, queryParams)
+      val searchCtx = SearchContext[Set, Id](jwt, parties, resolvedTemplateIds)
+      val source = SearchDb.cata(
```

Comment:
Can we use `search.search` instead? If we have an abstraction for the backend we should also use it and not pattern match on the backend here again.

Reply:
a689cf6552e